### PR TITLE
few add responders patches

### PR DESCRIPTION
**Pull request by joeyorlando**
# Which issue(s) this PR fixes

Closes https://github.com/grafana/support-escalations/issues/8143

Fix a few minor issues introduced in #3128:

- Fix slow `GET /users` internal API endpoint related to [this change](https://github.com/grafana/oncall/blob/dev/engine/apps/api/views/user.py#L239)
- Fix slow `GET /teams` internal API endpoint. Introduced a `short` query parameter that only invokes `apps.schedules.ical_utils.get_oncall_users_for_multiple_schedules` when `short=false`.
- Order results from `GET /teams` internal API endpoint by name (ascending)
- Fix search issue when searching for teams in the add responders popup window (this was strictly a frontend issue)
- CSS changes to add responders dropdown to fix lonnnggg results list:
  **Before**
  Screenshot 2023-10-31 at 10 06 20
  **After**
  Screenshot 2023-10-31 at 10 48 12

## Still todo

The `apps.schedules.ical_utils.get_oncall_users_for_multiple_schedules` method is still very slow when an instance has a lot of users (ex. `ops`). Ideally we should refactor this method to be more efficient because we still need to call this method under some circumstances. Ex. to populate this dropdown when Direct Paging a user (note that it didn't finish loading here on `ops`):
Screenshot 2023-10-30 at 18 14 59

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: engine/apps/api/views/user.py [hunk 1]
```diff
@@ -234,9 +234,27 @@ def schedules_with_oncall_users(self):
...
+        return self._is_currently_oncall_request() or self._is_short_oncall_request()
+
     def get_serializer_context(self):
         context = super().get_serializer_context()
-        context.update({"schedules_with_oncall_users": self.schedules_with_oncall_users})
+        context.update(
+            {
+                "schedules_with_oncall_users": self.schedules_with_oncall_users
+                if self._is_currently_oncall_or_short_request()
+                else {}
```
Comment (joeyorlando) [inline engine/apps/api/views/user.py:255]: main change, _only_ set `context["schedules_with_oncall_users"]` for certain requests. Previously we were _always_ setting it even if the serializer didn't need it (`UserLongSerializer` is the only serializer that needs this [here](https://github.com/grafana/oncall/pull/3220/files#diff-30309629484ad28e6fe09816e1bd226226d652ea977b6f3b6775976c729bf4b5R271-R272)).

File: engine/apps/api/views/user.py [hunk 2]
```diff
@@ -234,9 +234,27 @@ def schedules_with_oncall_users(self):
...
         return get_oncall_users_for_multiple_schedules(self.request.user.organization.oncall_schedules.all())
 
+    def _get_is_currently_oncall_query_param(self) -> str:
+        return self.request.query_params.get("is_currently_oncall", "").lower()
+
+    def _is_currently_oncall_request(self) -> bool:
+        return self._get_is_currently_oncall_query_param() in ["true", "false"]
+
+    def _is_short_oncall_request(self) -> bool:
+        return self.request.query_params.get("short", "true").lower() == "false"
```
Comment (matiasb) [inline engine/apps/api/views/user.py:244]: This seems confusing..? `_is_short_oncall_request` will return `True` if the `short` param value is `"false"`?

Comment (joeyorlando) [reply engine/apps/api/views/user.py:244]: ahh good catch, thanks 👍

**Review by matiasb — APPROVED** [review]

File: grafana-plugin/src/models/grafana_team/grafana_team.ts [hunk 3]
```diff
@@ -29,37 +31,32 @@ export class GrafanaTeamStore extends BaseStore {
...
   }
 
-  getSearchResult(query = '') {
-    if (!this.searchResult[query]) {
-      return [];
-    }
-
-    return this.searchResult[query].map((teamId: GrafanaTeam['id']) => this.items[teamId]);
+  getSearchResult() {
+    console.log('yoooo');
```
Comment (teodosii) [inline grafana-plugin/src/models/grafana_team/grafana_team.ts:59]: :rofl:

**Review by teodosii — APPROVED** [review]

File: grafana-plugin/src/models/grafana_team/grafana_team.ts (hunk 3 above)
Comment (joeyorlando) [reply grafana-plugin/src/models/grafana_team/grafana_team.ts:59]: oops lol, good catch 😆

**Pull request merged**